### DISCUSSION TITLE
Update Auro demo projects to be reliable resource

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#548] [Announcements]
## Problem

It has come to our attention that the Auro React Demo is significantly outdated in its development status. Furthermore, the existing demos have limited functionality and the setup instructions are inadequate.

## Recommended Solution

To address these issues effectively, the following updates are proposed for each demo repository:

1. Update the demo repository to align with the current Auro development standards, eliminating any legacy references and ensuring compliance with all relevant specifications.
2. Upgrade all dependencies to their latest versions to maintain compatibility and optimize performance.
3. Enhance documentation clarity and pragmatism, providing comprehensive setup instructions for new project configurations.

In addition to repository updates, the following functional enhancements are suggested:

1. Implement a robust test suite to validate updated elements, ensuring there are no visual or functional regressions upon installation.
2. Incorporate all Server-Side Rendering (SSR) compatible elements into the demo for comprehensive coverage.
3. Utilize functional data in the demos instead of static content to demonstrate element functionality effectively.
4. Adopt a single-page layout for the demo, emphasizing cleanliness and responsiveness to enhance user experience.
5. Update demos to use SSR build processes. 

## Expected Outcomes

- Improved alignment of demo repositories with current Auro development specifications.
- Enhanced compatibility and performance through upgraded dependencies.
- Clear and comprehensive documentation facilitating smoother project setup.
- Error-free installation of updated elements confirmed by robust test suite.
- Comprehensive coverage of SSR compatible elements in the demo.
- Effective demonstration of element functionality through functional data integration.
- Enhanced user experience with a clean and responsive single-page layout.

**Comment by fajar-apri-alaska** [comment]
Questions that I have right now:
Do we already have a direction for one or two of above points ?
or do we need to start researching some?

for example, using https://single-spa.js.org/docs/faq/#what-does-single-spa-do for the SPA-like apps for the demo.

**Reply by blackfalcon** [reply]
I am not super concerned with a SPA-like experience. The `Adopt a single-page layout for the demo...` statement is simply to say that all we need is a one-page document with all the components presented. No need for a complicated multi-page experience.
